frontend/packages/ux-editor/src/components/config/editModal/EditImage/EditImageUtils.test.ts

```diff
@@ -0,0 +1,101 @@
+import { componentMocks } from '@altinn/ux-editor/testing/componentMocks';
+import {
+  extractFileNameFromImageSrc,
+  updateComponentWithDeletedImageReference,
+  updateComponentWithImage,
+} from './EditImageUtils';
+import { ComponentType } from 'app-shared/types/ComponentType';
+
+describe('EditImageUtils', () => {
+  describe('updateComponentWithImage', () => {
+    it('updates component with new image source', () => {
+      const newImageSource = 'newImageSource';
+      const updatedComponent = updateComponentWithImage(
+        componentMocks[ComponentType.Image],
+        newImageSource,
+      );
+      expect(updatedComponent).toStrictEqual({
+        ...componentMocks[ComponentType.Image],
+        image: {
+          ...componentMocks[ComponentType.Image].image,
+          src: {
+            nb: newImageSource,
+          },
+        },
+      });
+    });
+  });
+
+  describe('updateComponentWithDeletedImageReference', () => {
+    it('updates component that already has an image source with deleted image source reference', () => {
+      const imageComponentWithImageSource = {
+        ...componentMocks[ComponentType.Image],
+        image: {
+          ...componentMocks[ComponentType.Image].image,
+          src: {
+            nb: 'someImageReference.png',
+          },
+        },
+      };
+      const updatedComponent = updateComponentWithDeletedImageReference(
+        imageComponentWithImageSource,
+      );
+      expect(updatedComponent).toStrictEqual({
+        ...imageComponentWithImageSource,
+        image: {
+          ...imageComponentWithImageSource.image,
+          src: {},
+        },
+      });
+    });
+
+    it('updates component that does not have an image source with deleted image source reference', () => {
+      const updatedComponent = updateComponentWithDeletedImageReference(
+        componentMocks[ComponentType.Image],
+      );
+      expect(updatedComponent).toStrictEqual({
+        ...componentMocks[ComponentType.Image],
+        image: {
+          ...componentMocks[ComponentType.Image].image,
+          src: {},
+        },
+      });
+    });
+  });
+
+  describe('extractFileNameFromImageSrc', () => {
+    it.each(['ttd/frontend-test', 'someRandomOrg/someRandomApp'])(
+      'extracts fileName from relative path when org/app is %s',
+      (orgApp: string) => {
+        const org = orgApp.split('/')[0];
+        const app = orgApp.split('/')[1];
+        const fileName = 'image.png';
+        const imageSource = `/${org}/${app}/${fileName}`;
+        const extractedFileName = extractFileNameFromImageSrc(imageSource, org, app);
+        expect(extractedFileName).toBe(fileName);
+      },
+    );
```

Comment:
Fin bruk av `each`! Men her ville jeg heller gjort testdaten til objekter, så slipper vi å bruke `split` til å ekstrahere den:
```suggestion
    type TestCase = {
      org: string;
      app: string;
    };
    const testCases: TestCase[] = [
      { org: 'ttd', app: 'frontend-test' },
      { org: 'someRandomOrg', app: 'someRandomApp' },
    ];
    it.each(testCases)(
      'extracts fileName from relative path when org/app is %s',
      ({org, app}) => {
        const fileName = 'image.png';
        const imageSource = `/${org}/${app}/${fileName}`;
        const extractedFileName = extractFileNameFromImageSrc(imageSource, org, app);
        expect(extractedFileName).toBe(fileName);
      },
    );
```

Reply:
Hm, ja, kan godt gjøre det, men det er ikke litt overkill for en test? Syns det var lettere å lese når det var mer kompakt

Reply:
Jeg synes det skurrer litt med `split` der. Her er et annet alternativ:
```suggestion
    it.each([
      ['ttd', frontend-test'],
      ['someRandomOrg', 'someRandomApp']
    ])(
      'extracts fileName from relative path when org is %s and app is %s',
      (org: string, app: string) => {
        const fileName = 'image.png';
        const imageSource = `/${org}/${app}/${fileName}`;
        const extractedFileName = extractFileNameFromImageSrc(imageSource, org, app);
        expect(extractedFileName).toBe(fileName);
      },
    );
```